decisions/0011-routes-ts.md

```diff
@@ -0,0 +1,80 @@
+# routes.ts
+
+Date: 2024-09-18
+
+Status: accepted
+
+## Context
+
+Now that Remix is being merged upstream into React Router, we have an opportunity to revisit our approach to route configuration for the Vite plugin.
+
+Remix ships with a default set of file system routing conventions. While convenient for demos, examples, tutorials and simple use cases, conventions like these come with a few major drawbacks:
+
+1. They're highly contentious.
+
+   File system routing conventions are inherently subjective, with numerous possible approaches and preferences. For example, some developers prefer flat route folder structures, while others prefer deeply nested directories. Some even prefer a mix of both. These differing opinions make it challenging to create a one-size-fits-all solution.
+
+   When we changed the default routing conventions between Remix v1 and v2, we found this caused friction for some users who didn't see the benefit in migrating and preferred the old conventions. While consumers could opt out of the new conventions, this may not have been obvious to everyone, and it made anyone using another convention feel like they were veering off the beaten path. In hindsight, we feel we could have been less opinionated in this area.
+
+2. Advanced usage gets convoluted.
+
+   Filenames are more limited compared to config-based routes since they have fewer characters to work with. They not only need to include the route path and any parameters—they also need to encode further route configuration, leading to increasingly elaborate naming conventions.
```

Comment:
Might be helpful to parenthetically provide some examples of the further route configuration needed (like pathless layouts, escaping special characters, etc.)